### PR TITLE
NPM publish workflow

### PR DESCRIPTION
**Pull request by supl**
This PR adds a GitHub Actions workflow to publish the node library to the NPM registry.

### CODE REVIEW

File: .github/npm-publish.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: NPM Registry Publication
+
+on:
+  push:
+    tags:
+      - node-[0-9]+.[0-9]+.[0-9]+
```
Comment (supl) [inline .github/npm-publish.yml:6]: The publishing workflow is triggered by pushing a tag like `node-1.2.0`

File: .github/npm-publish.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          node-version: ${{ matrix.node-version }}
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Build the distribution (from TypeScript to JavaScript)
+        run: npm run build
+
+      - name: Publish the package
+        run: npm publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (supl) [inline .github/npm-publish.yml:36]: I think Hiro's NPM_TOKEN is set already in the `scalar-labs` group on GitHub.
So no need to set up any other settings on GitHub.

File: node/.npmignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.eslintignore
+.eslintrc.json
+.prettierrc.js
+src/
```
Comment (supl) [inline node/.npmignore:4]: No need to include src/ since the package.json points to `dist/index.js` and `dist/index.d.ts`

**Review by ypeckstadt — APPROVED** [review]
LGTM.

File: .github/npm-publish.yml (hunk 2 above)
Comment (feeblefakie) [reply .github/npm-publish.yml:39]: It's set in the `scalardl-node-client-sdk` repo and it's not set in the organization secret. 
So, I think I need to set it to this repo, as well. Can you give me the one?

File: .github/npm-publish.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        working-directory: node
+
+    strategy:
+      matrix:
+        node-version:
+         - 16.x
```
Comment (feeblefakie) [inline .github/npm-publish.yml:19]: I think we need to define what version of node we support.
I think do we support only 14 for `scalardl-node-client-sdk`?

File: .github/npm-publish.yml (hunk 2 above)
Comment (supl) [reply .github/npm-publish.yml:39]: I am sorry. It's my mistake.
I thought NPM_TOKEN has been set as an organization secret already but it is CR_PAT.

> Can you give me the one?
This package will be published as a scoped package to the @scalar-labs organization in the NPM registry. The NPM_TOKEN will be your (feeblefakie) NPM access token.

Please check this document https://docs.npmjs.com/creating-and-viewing-access-tokens
(it seems the access token can be seen only once right after created though)

File: .github/npm-publish.yml (hunk 4 above)
Comment (supl) [reply .github/npm-publish.yml:19]: @feeblefakie 

> I think we need to define what version of node we support.
> I think do we support only 14 for scalardl-node-client-sdk?

Understood. Let's discuss about this.

scalardl-node-client-sdk now supports only Node.js version 16 (the active LTS).
(Thanks for your reminder. I noticed I have a typo [here](https://github.com/scalar-labs/scalardl-node-client-sdk/blob/master/package.json#L8). I will create another PR to fix it)

Comment (supl) [reply .github/npm-publish.yml:19]: > I think we need to define what version of node we support.

Discussed with Hiro.
We will support all active and maintenance LTS versions as long as the cost isn't too high (for example, no need to have additional codes for supporting various versions)

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**